### DISCUSSION TITLE
Adding Debug Information

### DISCUSSION THREAD

**Discussion by corbin-poteet** [Sixze/ALS-Refactored#58] [Q&A]
I'd like to add additional debug information for things like the player's inventory, but there's like zero information online on how to use Unreal's DisplayDebug functionality. How would you go about adding additional information to the debug screen in child classes?

Thank you so much for everything you do by the way, this project is amazing.

**Comment by Sixze** [comment]
You can override the `AAlsCharacter::DisplayDebug()` function in your child class and do whatever you want inside. Take a look at the `AlsCharacter_Debug.cpp` file to see how the existing debug functionality is made.

I don't remember having to look for information about DisplayDebug online, basically I just looked at how it's used in the source code of the engine and did it in the image and likeness.

**Reply by corbin-poteet** [reply]
How are different Debug Displays toggled on and off, like with shift + 1, shift + 2, etc? I've scoured the code for the logic that handles that but I can't seem to find it

**Reply by Alex-G** [reply]
Look here: https://github.com/Sixze/ALS-Refactored/blob/main/Config/DefaultInput.ini
